docs/design/proposals/destructive-operations/destructive-operations.md

```diff
@@ -0,0 +1,130 @@
+# Semantics of destructive operations
+
+## Summary
+
+This design document stems from the initial discussion on the [original Github issue #82](https://github.com/aws-controllers-k8s/community/issues/82).
+
+When deleting an ACK custom resource, the underlying AWS resource is first deleted. Users should expect this to be the default behaviour, as creating and updating CR similarly have the same respective actions in AWS. However users have requested that some resources should continue to live on after the ACK CR has been deleted. Resources with stateful data, such as S3 buckets, RDS instance or EBS volumes, may want to be retained so that they may be migrated between ACK installations or off ACK entirely without requiring they be backed up, deleted and then restored.
+
+Similar to the CDK[`RemovalPolicy` type](https://docs.aws.amazon.com/cdk/api/v2//docs/aws-cdk-lib.RemovalPolicy.html#members) and the native [K8s PV reclaim policy](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reclaiming), ACK should support the ability to leave the AWS resource alive while still deleting the custom resource object. 
```

Comment:
I'll be honest here... any time I see a reference to CloudFormation's (this is CDK, yes, but it's all CloudFormation...) [DeletionPolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html), I shudder. I really despise the entire user experience of *configuring some tool to behave different ways when the user takes a specific action* (the action in this case is the `DELETE` action.  This is a non-obvious interface and I am opposed to it. Instead, we should encourage a user experience that requires the user to explicitly call a *different action* in order to have the tool behave differently.

Reply:
After chatting with @RedbackThomson a bit more about this, I'm going to take a slightly less rigid approach to this. Kubernetes API unfortunately does not have a "control channel" for which to send *instructions* to a *controller* about how to handle a particular *action*. Instead, we have the `metadata.annotations` Hammer which is unfortunate IMHO because this is on the *resource* and not where it should be: as decorative data on the *request*.

Alas, the `metadata.annotations` Hammer is all we have so I will reconsider this proposal in light of that and see if I can come up with a variation that doesn't offend my sense of poor coupling.